packages/mui-base/src/Composite/Root/useCompositeRoot.ts

```diff
@@ -130,31 +166,54 @@ export function useCompositeRoot(params: UseCompositeRootParameters) {
                     // eslint-disable-next-line no-nested-ternary
                     event.key === ARROW_DOWN ? 'bl' : event.key === ARROW_RIGHT ? 'tr' : 'tl',
                   ),
+                  rtl: isRtl,
                 },
               )
             ] as number; // navigated cell will never be nullish
           }
 
-          const toEndKeys = {
-            horizontal: [ARROW_RIGHT],
-            vertical: [ARROW_DOWN],
-            both: [ARROW_RIGHT, ARROW_DOWN],
-          }[orientation];
+          const toEndKeys = isRtl
+            ? {
+                horizontal: [ARROW_LEFT],
+                vertical: [ARROW_DOWN],
+                both: [ARROW_LEFT, ARROW_DOWN],
+              }[orientation]
+            : {
+                horizontal: [ARROW_RIGHT],
+                vertical: [ARROW_DOWN],
+                both: [ARROW_RIGHT, ARROW_DOWN],
+              }[orientation];
```

Comment:
This could be a bit simpler by assigning `const horizontalEndKey = isRtl ? ARROW_LEFT ? ARROW_RIGHT` and same for `start`